src/Modules/OrchardCore.Commerce/Handlers/UniqueSkuValidationHandler.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Lombiq.HelpfulLibraries.OrchardCore.Contents;
+using OrchardCore.Commerce.Models;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Handlers;
+using OrchardCore.DisplayManagement.ModelBinding;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using YesSql;
+
+namespace OrchardCore.Commerce.Handlers;
+
+public class UniqueSkuValidationHandler : ContentPartHandler<ProductPart>
+{
+    private readonly ISession _session;
+    private readonly IUpdateModelAccessor _updateModelAccessor;
+
+    public UniqueSkuValidationHandler(
+        IContentManager contentManager,
+        ISession session,
+        IUpdateModelAccessor updateModelAccessor)
+    {
+        _session = session;
+        _updateModelAccessor = updateModelAccessor;
+    }
+
+    public override async Task UpdatedAsync(UpdateContentContext context, ProductPart instance)
+    {
+        var products = (await _session
+            .QueryContentItem(PublicationStatus.Published)
+            .ListAsync()).AsList();
+
+        var isNotUnique = products
+            .Any(sku =>
+                sku.As<ProductPart>()?.Sku == instance.Sku &&
+                sku.ContentItem.ContentItemId != instance.ContentItem.ContentItemId);
```

Comment:
Can't you do this in the query? Please check if there is an index you can use to filter this DB side. If not, then create it.